project_recalculate/models/project_task.py

```diff
@@ -0,0 +1,263 @@
+                                   calendar=None):
+        calendar_id = calendar.id if calendar else None
+        resource_id = resource.id if resource else None
+        default_interval = self._interval_default_get()
+        rc = self.pool['resource.calendar']
+        intervals = rc.get_working_intervals_of_day(
+            self.env.cr, self.env.uid, calendar_id, start_dt=day_date,
+            compute_leaves=True, resource_id=resource_id,
+            default_interval=default_interval, context=self.env.context)
+        return intervals and intervals[0] or False
+
+    def _calendar_schedule_days(self, days, day_date,
+                                resource=None, calendar=None):
+        if not day_date:
+            return (False, False)
+        calendar_id = calendar.id if calendar else None
+        resource_id = resource.id if resource else None
+        default_interval = self._interval_default_get()
+        intervals = self.pool['resource.calendar'].schedule_days(
+            self.env.cr, self.env.uid, calendar_id, days, day_date=day_date,
+            compute_leaves=True, resource_id=resource_id,
+            default_interval=default_interval, context=self.env.context)
+        return (intervals and intervals[0] or False,
+                intervals and intervals[-1] or False)
+
+    @api.multi
+    def task_recalculate(self):
+        """
+            Recalculate task start date and end date depending on
+            project calculation_type, estimated_days and from_days
+        """
+        to_string = fields.Datetime.to_string
+        for task in self:
+            if not task.include_in_recalculate:
+                continue
+            resource, calendar = task._resource_calendar_select()
+            increment, project_date, from_days = task._calculation_prepare()
+            date_start = False
+            date_end = False
+            from_days = self._from_days_dec(
+                from_days, project_date, resource, calendar, increment)
+            start = self._calendar_schedule_days(
+                from_days, project_date, resource, calendar)[1]
+            if start:
+                day = start[0].replace(hour=0, minute=0, second=0)
+                first = self._first_interval_of_day_get(
+                    day, resource, calendar)
+                if first:
+                    date_start = first[0]
+            if date_start:
+                end = self._calendar_schedule_days(
+                    task.estimated_days, date_start, resource, calendar)[1]
+                if end:
+                    date_end = end[1]
+            task.with_context(task.env.context, task_recalculate=True).write({
+                'date_start': date_start and to_string(date_start) or False,
+                'date_end': date_end and to_string(date_end) or False,
+                'date_deadline': date_end and to_string(date_end) or False,
+            })
+        return True
+
+    @api.one
```

Comment:
Here I'm calling to ```_dates_onchange``` that calculate estimated_days and from_days from date_start an date_end of the task. And have to read also project date, calendar, etc ...
I can not suppuse that write will call with a multiple recordset of tasks with the same dates, same project, calendar, ...
So, in this point of inherit chain a need to demultiplexate writing, althougth performance is affecting.